src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSource.java

```diff
@@ -174,20 +175,12 @@ public ReadsSparkSource(final JavaSparkContext ctx, final ValidationStringency v
      * Loads the header using Hadoop-BAM.
      * @param filePath path to the bam.
      * @param referencePath Reference path or null if not available. Reference is required for CRAM files.
-     * @param auth authentication information if using GCS.
      * @return the header for the bam.
      */
-    public SAMFileHeader getHeader(final String filePath, final String referencePath, final AuthHolder auth) {
+    public SAMFileHeader getHeader(final String filePath, final String referencePath) {
         // GCS case
         if (BucketUtils.isCloudStorageUrl(filePath)) {
-            try {
-                Storage.Objects storageClient = auth.makeStorageClient();
-                try (final SamReader reader = BAMIO.openBAM(storageClient, filePath, validationStringency)) {
-                    return reader.getFileHeader();
-                }
-            } catch (Exception e) {
-                throw new UserException("Failed to read bam header from " + filePath + "\n Caused by:" + e.getMessage(), e);
-            }
+            return new ReadsDataSource(IOUtils.getPath(filePath)).getHeader();
```

Comment:
Wrap the construction of the `ReadsDataSource` within a try-with-resources block to ensure it's closed before we return the header.